### ISSUE TITLE
Allow complex Jobs

### ISSUE THREAD

**Issue by Eibriel** [armadillica/flamenco#88]
Old:

Allow the creation of complex Jobs with multiple tasks, the tasks could depend on each other.
For example a Job could have two Tasks: render and composite. The Composite task depends on the render task completion. The tasks could have (or not) output images for Thumbnailing.

The Dashboard should show each task for each job. Showing the progress of each Task.
Each task should have an option to pause, stop, skip or restart.
Each job should have an option to pause, stop, skip or restart.
Must be possible to restart the jobs or tasks marked as failed. And the Server should have the ability to remember which tasks failed in which Worker to avoid them.
A job or task should be marked as failed after a couple of attempts (configurable).

The Workers view should show the task or tasks assigned to each Worker. And should allow to Deactivate a Worker or made a Worker skip a given task.

**Comment by Eibriel** [comment]
* Allow Workers to execute any Command
* Make the Worker?Manager?Server? parse the output, the Thumbnailer code will be in the parser.
* Allow the Server to receive Thumbnails.
* Make the parser send Thumbnails to Server.
* Save Jobs and Subjobs in database.
* Add type to the Subjobs.
* Find a way to store different Jobs definitions (compilers). Taking input parameters and building jobs and commands.

**Comment by fsiddi** [comment]
Hey there,
this was successfully completed, but I think you mentioned something about it being broken due to the new task table locking system strategy. Could you please let me know if everything is fine and we can archive this @Eibriel?

Thanks!

**Comment by eliemichel** [comment]
Has the possibility of a task depending on another one been implemented?

**Comment by Eibriel** [comment]
@eliemichel yes, but you need to write a custom Job Compiler, for example:
https://github.com/armadillica/flamenco/blob/master/flamenco/server/application/job_compilers/tiled_blender_render.py

Here you can see how `create_task` works https://github.com/armadillica/flamenco/blob/798d9cb54c2d152ad236216e5402cc379602c866/flamenco/server/application/modules/tasks/__init__.py#L70

**Comment by Eibriel** [comment]
@fsiddi yes we can archive this!

**Comment by eliemichel** [comment]
@Eibriel thx!
So the dependency system is based on `child_id`. But does that mean that there can be only one task requiring this one? I would have expected it to be more something like `require_ids` and a list of ids of tasks that must finish before.

**Comment by Eibriel** [comment]
Yes, for the moment you can only have one child with many parents.
